### PR TITLE
BUGFIX: Make URI suffix configurable for the ESFrontendNode RoutePart

### PR DESCRIPTION
**Pull request by nezaniel**
This makes the uri suffix (e.g. .html) configurable on route (part) level, which is essential for Neos.SEO's XML sitemap and robots.txt as well as custom formats for rendering nodes as pdf etc.

**Upgrade instructions**

none

**Review instructions**

none

**Checklist**

- [x] Code follows the PSR-12 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by dlubitz** [comment]
Looks like a duplicate of https://github.com/neos/neos-development-collection/pull/5189

**Review by dlubitz — APPROVED** [review]

**Comment by nezaniel** [comment]
yes, but without the other changes

**Review by mhsdesign — APPROVED** [review]
Looks good ;)

it seems this got just lost in the rewrite.
In Neos 8 it looked like such: https://github.com/neos/neos-development-collection/blob/072186e1601d82a8565942a05c9fe7253cf7f85d/Neos.Neos/Classes/Routing/FrontendNodeRoutePartHandler.php#L296

We were only using the options as it was only a single cr and we configured the default via `defaultUriSuffix` globally.

**Comment by mhsdesign** [comment]
But this is only half of the fix!!!

in 8.3 we also have other options aside from `uriPathSuffix` which are used in Neos Seo!!! https://github.com/neos/neos-seo/blob/c2205573caac5f6a2b37dec4651ad0d336b4f31d/Configuration/Routes.yaml#L18

`onlyMatchSiteNodes`:
 - Whether the current route part should only match/resolve site nodes (e.g. the homepage)

`nodeType`:
- Whether the given $node is allowed according to the "nodeType" option

**Review by mhsdesign — CHANGES_REQUESTED** [review]
see above, its only 50% of the rent ...

**Comment by nezaniel** [comment]
we ignored onlyMatchSiteNodes for now because my-site/some-path/robots.txt does not 404 as expected but does not lie either. It would require a larger effort to implement it and is postponed to later

As with suffix vs. CR: we discussed this and came to the conclusion that we are fine with global overrides on route level for now, as CR specific overrides are no requirement yet. This as well can be added later if necessary.

In conclusion: This change is already sufficient for Neos.Seo 4.1.4

**Review by mhsdesign — APPROVED** [review]

**Comment by mhsdesign** [comment]
Perfekt, i just wanted it to bring that to attention. Then lets have it. Fine by reading ;)
